dynamodb/src/main/scala/monix/connect/dynamodb/DynamoDbSubscriber.scala

```diff
@@ -18,44 +18,64 @@
 package monix.connect.dynamodb
 
 import monix.eval.Task
+import monix.execution.Ack.Continue
+import monix.execution.atomic.AtomicInt
 import monix.execution.cancelables.AssignableCancelable
 import monix.execution.{Ack, Callback, Scheduler}
-import monix.reactive.observers.Subscriber
 import monix.reactive.Consumer
 import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
 import software.amazon.awssdk.services.dynamodb.model.{DynamoDbRequest, DynamoDbResponse}
+import monix.reactive.observers.{SafeSubscriber, Subscriber}
 
+import scala.compat.java8.FutureConverters._
 import scala.concurrent.Future
 
-private[dynamodb] class DynamoDbSubscriber[In <: DynamoDbRequest, Out <: DynamoDbResponse]()(
+private[dynamodb] class DynamoDbSubscriber[In <: DynamoDbRequest, Out <: DynamoDbResponse](retriesPerOp: Int)(
   implicit
   dynamoDbOp: DynamoDbOp[In, Out],
   client: DynamoDbAsyncClient)
   extends Consumer[In, Out] {
 
+  require(retriesPerOp >= 1, "Retries per operation must be higher or equal than 1.")
+
   override def createSubscriber(cb: Callback[Throwable, Out], s: Scheduler): (Subscriber[In], AssignableCancelable) = {
-    val sub = new Subscriber[In] {
+    val safeSubscriber = SafeSubscriber {
+      new Subscriber[In] {
 
-      implicit val scheduler = s
-      private var dynamoDbResponse: Task[Out] = _
+        implicit val scheduler = s
+        private var dynamoDbResponse: Task[Out] = _
 
-      def onNext(dynamoDbRequest: In): Future[Ack] = {
-        dynamoDbResponse = Task.from(dynamoDbOp.execute(dynamoDbRequest))
+        def onNext(dynamoDbRequest: In): Future[Ack] = {
+          val retries: AtomicInt = AtomicInt(retriesPerOp)
 
-        dynamoDbResponse.onErrorRecover { case _ => monix.execution.Ack.Stop }
-          .map(_ => monix.execution.Ack.Continue)
-          .runToFuture
-      }
+          dynamoDbResponse = Task.from(dynamoDbOp.execute(dynamoDbRequest))
+          dynamoDbResponse
+            .onErrorHandleWith(ex =>
+              Task.defer {
+                if (retries.get() > 1) {
+                  retries.set(retries.get() - 1)
+                  Task.deferFuture(toScala(dynamoDbOp.execute(dynamoDbRequest)))
+                } else {
+                  Task.raiseError(ex)
+                }
+              })
+            .redeem(ex => {
+              onError(ex)
+              monix.execution.Ack.Stop
+            }, _ => Continue)
+            .runToFuture
+        }
 
-      def onComplete(): Unit = {
-        dynamoDbResponse.runAsync(cb)
-      }
+        def onComplete(): Unit = {
+          dynamoDbResponse.runAsync(cb)
```

Comment:
Should `dynamoDbResponse` run twice if there is `onComplete` or did you min to signal completion of "main" `Task`?

I think you could keep the last response in `var`, like `FoldLeftTaskConsumer`